constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/ForEachActivitySpans.java

```diff
@@ -1,34 +1,35 @@
 package gov.nasa.jpl.aerie.constraints.tree;
 
+import gov.nasa.jpl.aerie.constraints.model.ActivityInstance;
 import gov.nasa.jpl.aerie.constraints.model.EvaluationEnvironment;
 import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
 import gov.nasa.jpl.aerie.constraints.time.Interval;
 import gov.nasa.jpl.aerie.constraints.time.Spans;
+import org.apache.commons.lang3.function.TriFunction;
 
 import java.util.HashMap;
 import java.util.Objects;
 import java.util.Set;
 
-public final class ForEachActivitySpans implements Expression<Spans> {
-  public final String activityType;
-  public final String alias;
-  public final Expression<Spans> expression;
+public record ForEachActivitySpans(
```

Comment:
Big fan of using records for constraint AST nodes 🚀

As I write this, I do wonder what the implications are for equality of these `TriFunctions` 🤔

Reply:
This is the doc comment on `ForEachActivitySpans.MatchType`:

```
   * A helper class for activity predicates that only match on type.
   *
   * This exists to provide an equality check between predicates. You could make an anonymous function that implements
   * {@link TriFunction}, but tests will fail because all such objects are considered non-equal, even when in reality they
   * have identical behavior.```

I can make a similar record for matching on type and arguments, but currently we don't do any equality check on those types of nodes, so it isn't necessary.